src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -1,281 +1,338 @@
-                        }
-
-                        return this._getStripeJs().createPaymentMethod('card', this._cardElement, this._mapStripePaymentMethodOptions())
-                            .then(stripePaymentMethod => {
-                                if (stripePaymentMethod.error || !stripePaymentMethod.paymentMethod.id) {
-                                    throw new PaymentMethodFailedError(stripePaymentMethod.error && stripePaymentMethod.error.message);
-                                }
-
-                                if (!paymentIntent) {
-                                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
-                                }
-
-                                const stripeCardPaymentOptions = {
-                                    ...this._mapStripeCardPaymentOptions(shouldSaveInstrument),
-                                    payment_method: stripePaymentMethod.paymentMethod.id,
-                                };
-
-                                return this._getStripeJs().handleCardPayment(paymentIntent, stripeCardPaymentOptions);
-                            })
-                            .then(stripeResponse => {
-                                if (stripeResponse.error || !stripeResponse.paymentIntent.id) {
-                                    throw new PaymentMethodFailedError(stripeResponse.error && stripeResponse.error.message);
-                                }
-
-                                const paymentPayload = {
-                                    methodId: payment.methodId,
-                                    paymentData: {
-                                        nonce: stripeResponse.paymentIntent.id,
-                                        shouldSaveInstrument,
-                                    },
-                                };
-
-                                return this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
-                            });
-                    });
-            });
+        return await this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload))
+            .catch(error => this._processAdditionalAction(error));
     }
 
     finalize(): Promise<InternalCheckoutSelectors> {
         return Promise.reject(new OrderFinalizationNotRequiredError());
     }
 
     deinitialize(): Promise<InternalCheckoutSelectors> {
-        if (this._cardElement) {
-            this._cardElement.unmount();
-        }
+        this._unmountElement();
 
         return Promise.resolve(this._store.getState());
     }
 
-    private _getStripeJs(): StripeV3Client {
-        if (!this._stripeV3Client) {
-            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+    private async _processAdditionalAction(error: unknown): Promise<any> {
+        if (!(error instanceof RequestError) || !some(error.body.errors, {code: 'additional_action_required'})) {
```

Comment:
same as above, you could make invert the condition to make it easier to read
```js
const isAdditionalActionError = error instanceof RequestError && 
  some(error.body.errors, {code: 'additional_action_required'});

if (!isAdditionalActionError) {
```

You could even improve the types here and create a new type AdditionalErrorAction

```function isAdditionalActionError(error: unknown): error is AdditionalErrorAction```